drivers/gpio/gpio_rts5912.c

```diff
@@ -0,0 +1,457 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2024 Realtek Semiconductor Corporation, SIBG-SD7
+ * Author: Lin Yu-Cheng <lin_yu_cheng@realtek.com>
+ */
+
+#define DT_DRV_COMPAT realtek_rts5912_gpio
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/irq.h>
+#include "zephyr/drivers/gpio/gpio_utils.h"
+#include <zephyr/logging/log.h>
+#include <zephyr/dt-bindings/gpio/realtek-gpio.h>
+
+#include <reg/reg_gpio.h>
+
+LOG_MODULE_REGISTER(gpio_rts5912, CONFIG_GPIO_LOG_LEVEL);
+
+#define RTS5912_GPIOA_REG_BASE ((GPIO_Type *)(DT_REG_ADDR(DT_NODELABEL(gpioa))))
+
+struct gpio_rts5912_config {
+	struct gpio_driver_config common;
+	volatile uint32_t *reg_base;
+	uint8_t num_pins;
+};
+
+struct gpio_rts5912_data {
+	struct gpio_driver_data common;
+	sys_slist_t callbacks;
+};
+
+static int pin_is_valid(const struct gpio_rts5912_config *config, gpio_pin_t pin)
+{
+	if (pin >= config->num_pins) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int pin_output_high(const struct device *port, gpio_pin_t pin)
+{
+	const struct gpio_rts5912_config *config = port->config;
+	volatile uint32_t *gcr = &config->reg_base[pin];
+
+	int err = pin_is_valid(config, pin);
+
+	if (err) {
+		return err;
+	}
+
+	if (*gcr & GPIO_GCR_OUTMD_Msk) {
+		/* Switch I/O mode to input mode when configuration is open-drain with output high
+		 */
+		*gcr = ((*gcr & ~GPIO_GCR_DIR_Msk) | GPIO_GCR_OUTCTRL_Msk);
+	} else {
+		*gcr |= (GPIO_GCR_OUTCTRL_Msk | GPIO_GCR_DIR_Msk);
```

Comment:
redundant outer parenthesis, drop

Reply:
removed it and have check the same problem in this .c file